core-backend/sandbox/src/env.rs

```diff
@@ -100,15 +101,15 @@ impl<E: Ext + IntoExtInfo + 'static> Environment<E> for SandboxEnvironment<E> {
         memory_pages: &BTreeMap<PageNumber, Box<PageBuf>>,
         mem_size: WasmPageNumber,
     ) -> Result<Self, BackendError<Self::Error>> {
-        let ext_carrier = ExtCarrier::new(ext);
+        let ext_carrier = ExtCarrier::new(ErrorSavingExt::new(ext));
 
         let mem: DefaultExecutorMemory = match SandboxMemory::new(mem_size.0, None) {
             Ok(mem) => mem,
             Err(e) => {
                 return Err(BackendError {
                     reason: SandboxEnvironmentError::CreateEnvMemory,
                     description: Some(format!("{:?}", e).into()),
-                    gas_amount: ext_carrier.into_inner().into_gas_amount(),
+                    gas_amount: ext_carrier.into_inner().inner.into_gas_amount(),
```

Comment:
```suggestion
                    gas_amount: ext_carrier.into_inner().into_gas_amount(),
```

`into_inner().inner` seems extraordinary strange to me :D
 Can we deal with it somehow?

Reply:
Impl `IntoExtInfo` for `ErrorSavingExt`